extensions/wikia/AnalyticsEngine/js/universal_analytics.js

```diff
@@ -4,473 +4,473 @@
-		} else {
-			// prevent errors when wgNoExternals is set or user opt-outs
-			window.ga = function () {
-			};
-		}
-
-		var cookieExists,
-			isProductionEnv,
-			blockingTracked = {
-				babDetector: false,
-				pageFair: false
-			},
-			GASettings = {
-				babDetector: {
-					trackName: 'babDetector',
-					dimension: 6,
-					name: 'babdetector'
-				},
-				pageFair: {
-					trackName: 'pageFair',
-					dimension: 7,
-					name: 'pagefair'
-				}
-			},
-			listenerSettings = [
-				{
-					eventName: 'bab.blocking',
-					value: true,
-					detectorSettings: GASettings.babDetector
-				},
-				{
-					eventName: 'bab.not_blocking',
-					value: false,
-					detectorSettings: GASettings.babDetector
-				},
-				{
-					eventName: 'pf.blocking',
-					value: true,
-					detectorSettings: GASettings.pageFair
-				},
-				{
-					eventName: 'pf.not_blocking',
-					value: false,
-					detectorSettings: GASettings.pageFair
-				}
-			];
-
+	if (!window.wgNoExternals) {
 		/**
-		 * Main Tracker
+		 * Creates a temporary global ga object and loads analy  tics.js.
+		 * Paramenters o, a, and m are all used internally.  They could have been declared using 'var',
+		 * instead they are declared as parameters to save 4 bytes ('var ').
 		 *
-		 * To be used for everything that is not advertisement
+		 * @param {Window}      i The global context object.
+		 * @param {Document}    s The DOM document object.
+		 * @param {string}      o Must be 'script'.
+		 * @param {string}      g URL of the analytics.js script. Inherits protocol from page.
+		 * @param {string}      r Global name of analytics object.  Defaults to 'ga'.
+		 * @param {DOMElement?} a Async script tag.
+		 * @param {DOMElement?} m First script tag in document.
 		 */
-		isProductionEnv = !window.wgGaStaging;
+		(function (i, s, o, g, r, a, m) {
+			i['GoogleAnalyticsObject'] = r; // Acts as a pointer to support renaming.
 
-		cookieExists = function (cookieName) {
-			return document.cookie.indexOf(cookieName) > -1;
-		};
-
-		// Main Roll-up Account - UA-32129070-1/UA-32129070-2
-		if (isProductionEnv) {
-			// Production Environment
-			window.ga(
-				'create', 'UA-32129070-1', 'auto',
-				{
-					'sampleRate': 100,
-					'allowLinker': true,
-					'userId': window.wgGAUserIdHash
+			// Creates an initial ga() function.  The queued commands will be executed once analytics.js loads.
+			i[r] = i[r] || function () {
+				(i[r].q = i[r].q || []).push(arguments);
+			},
 
+				// Sets the time (as an integer) this tag was executed.  Used for timing hits.
+				i[r].l = 1 * new Date();
+
+			// Insert the script tag asynchronously.  Inserts above current tag to prevent blocking in
+			// addition to using the async attribute.
+			a = s.createElement(o),
+				m = s.getElementsByTagName(o)[0];
+			a.async = 1;
+			a.src = g;
+			m.parentNode.insertBefore(a, m);
+		})(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');
```

Comment:
is it OK to always load GA script itself?

Reply:
IMO it's fine as we always want to track all users opted-in/opted-out.